app/src/main/java/com/zeapo/pwdstore/crypto/PasswordCreationActivity.kt

```diff
@@ -279,166 +279,179 @@ class PasswordCreationActivity : BasePgpActivity(), OpenPgpServiceConnection.OnB
-        if (gpgIdentifierFile == null) {
-            snackbar(message = resources.getString(R.string.failed_to_find_key_id))
-            return@with
-        }
-        val gpgIdentifierFileContent = gpgIdentifierFile.useLines { it.firstOrNull() } ?: ""
-        when (val identifier = parseGpgIdentifier(gpgIdentifierFileContent)) {
-            is GpgIdentifier.KeyId -> data.putExtra(OpenPgpApi.EXTRA_KEY_IDS, arrayOf(identifier.id))
-            is GpgIdentifier.UserId -> data.putExtra(OpenPgpApi.EXTRA_USER_IDS, arrayOf(identifier.email))
-            null -> {
-                snackbar(message = resources.getString(R.string.invalid_gpg_id))
+            val data = receivedIntent ?: Intent()
+            data.action = OpenPgpApi.ACTION_ENCRYPT
+
+            // pass enters the key ID into `.gpg-id`.
+            val repoRoot = PasswordRepository.getRepositoryDirectory(applicationContext)
+            val gpgIdentifierFile = File(repoRoot, directory.text.toString()).findTillRoot(".gpg-id", repoRoot)
+            if (gpgIdentifierFile == null) {
+                snackbar(message = resources.getString(R.string.failed_to_find_key_id))
                 return@with
             }
-        }
-        data.putExtra(OpenPgpApi.EXTRA_REQUEST_ASCII_ARMOR, true)
-
-        val content = "$editPass\n$editExtra"
-        val inputStream = ByteArrayInputStream(content.toByteArray())
-        val outputStream = ByteArrayOutputStream()
-
-        val path = when {
-            // If we allowed the user to edit the relative path, we have to consider it here instead
-            // of fullPath.
-            directoryInputLayout.isEnabled -> {
-                val editRelativePath = directory.text.toString().trim()
-                if (editRelativePath.isEmpty()) {
-                    snackbar(message = resources.getString(R.string.path_toast_text))
-                    return
-                }
-                val passwordDirectory = File("$repoPath/${editRelativePath.trim('/')}")
-                if (!passwordDirectory.exists() && !passwordDirectory.mkdir()) {
-                    snackbar(message = "Failed to create directory ${editRelativePath.trim('/')}")
-                    return
+            val gpgIdentifierFileContent = gpgIdentifierFile.useLines { it.firstOrNull()?.trim() } ?: ""
+            if (gpgIdentifierFileContent.isEmpty()) {
+                registerForActivityResult(StartActivityForResult()) { result ->
+                    if (result.resultCode == RESULT_OK) {
+                        result.data?.getStringExtra(OpenPgpApi.EXTRA_KEY_ID)?.let { keyId ->
+                            gpgIdentifierFile.writeText(keyId)
+                            encrypt(data)
+                        }
+                    }
+                }.launch(Intent(this@PasswordCreationActivity, GetKeyIdsActivity::class.java))
+                return@with
+            }
```

Comment:
What's happening here is that if the `.gpg-id` file is empty, we launch `GetKeyIdsActivity` expecting a String key from it. When we get this key, it is written into the `.gpg-id` file we initially found. TODO: commit this diff?